proton-c/bindings/python/proton/utils.py

```diff
@@ -132,10 +132,12 @@ def __init__(self, connection, receiver, fetcher, credit=1):
             raise LinkException("Failed to open receiver %s, source does not match" % self.link.name)
         if credit: receiver.flow(credit)
         self.fetcher = fetcher
+        self.container = connection.container
 
     def __del__(self):
         self.fetcher = None
-        self.link.handler = None
+        self.link.handler = None  # fails if reactor finalizes first
```

Comment:
What happens with the error if it fails?  I suppose in that case it doesn't matter that the subsequent statement is not reached?